src/bot.py

```diff
@@ -146,8 +143,17 @@ def main(self):
     try:
         TwitterBot = Bot()
         TwitterBot.main()
+        
+        
+        
+    except tweepy.RateLimitError:
+        logging.critical("Tweeting failed due to ratelimit. Waiting {} more minutes.".format(cooldown))
+        time.sleep(cooldown)

```

Comment:
Please review the logic here. Nothing will happen after the `time.sleep(cooldown)` function.

Maybe it would be best to move both handlers at the function which handles the tweets for now.

Reply:
That is what I came up, after I tested the bot. We should keep the authentication error handler at the main function, and the RateLimit and generic Exception handler at the bot.

Reply:
The above approach is not working properly. I am keeping the generic exception handler at the main function to avoid ambiguity.